inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -248,15 +247,13 @@ public function get_sitemap_links( $type, $max_entries, $current_page ) {
 				continue;
 			}
 
-			$posts_to_exclude = explode( ',', $options['excluded-posts'] );
-
 			foreach ( $posts as $post ) {
 
-				if ( WPSEO_Meta::get_value( 'meta-robots-noindex', $post->ID ) === '1' ) {
+				if ( in_array( $post->ID, $posts_to_exclude, true ) ) {
```

Comment:
Strict compare could be tricky because `$post->ID` is string. You can see more details - #8787.